app/nodejs/process.js

```diff
@@ -20,15 +20,154 @@ process.js - take data in raw_data bucket, process,
+      `to ${config.PROCESSED_DATA_BUCKET}`);
+
+  await storage
+      .bucket(config.RAW_DATA_BUCKET)
+      .file(config.RAW_DATA_FILE)
+      .download({destination: tempDataFile});
+
+  logger.info(`downloadRawData: downloaded data to ${tempDataFile}`);
+
+  return tempDataFile;
+}
+/**
+ * Process local file, producing aggregate data
+ * @param {string} tempDataFile Local filename containing downloaded data
+ * @return {object} Correlated aggregate data
+ */
+async function processRawData(tempDataFile) {
+  logger.info('processRawData: start processing data');
+
+  return new Promise((resolve, reject) => {
+    const aggregate = {};
+    let ignoredRecords = 0;
+    let countedRecords = 0;
+    const results = [];
+
+
+    fs.createReadStream(tempDataFile)
+        .pipe(csv())
+        .on('data', (data) => results.push(data))
+        .on('end', () => {
+          results.forEach(function(row) {
+            let validRow = true;
+
+            config.FACETS.forEach(function(facet) {
+              if ((row[facet] == '') || (row[facet] == '?')) {
+                ignoredRecords += 1;
+                validRow = false;
+              }
+            });
+
+            if (validRow) {
+            // build aggregate identifier
+              const rowKeyValues = [];
+              config.FACETS.forEach(function(facet) {
+                rowKeyValues.push(row[facet]);
+              });
+              const rowKey = rowKeyValues.join('/');
+
+              // Build the base data structure on first interaction
+              if (!aggregate[rowKey]) {
+                aggregate[rowKey] = {'_counter': 0};
+                config.SEGMENTS.forEach(function(segment) {
+                  if (!aggregate[rowKey][segment]) {
+                    aggregate[rowKey][segment] = 0;
+                  }
+                });
+              }
+
+              // Record the relevant data
+              config.SEGMENTS.forEach(function(segment) {
+                if (row[segment] == 'true') {
+                  aggregate[rowKey][segment] += 1;
+                }
+              });
+
+              // Increment counters
+              aggregate[rowKey]['_counter'] += 1;
+              countedRecords += 1;
+            }
+          });
+
+          logger.info(`processRawData: processed ${countedRecords} ` +
+          `records, removed ${ignoredRecords}`);
+
+          resolve(aggregate);
+        });
+  });
+}
+
+
+/**
+ * Write aggregate data to Cloud Storage
+ * @param {object} aggregate Aggregated data
+ */
+function writeProcessedData(aggregate) {
+  logger.info('writeProcessedData: start writing');
+
+  let counter = 0;
+
+  const processedBucket = storage.bucket(config.PROCESSED_DATA_BUCKET);
+
+  const writeData = new Promise((resolve, reject) => {
+    Object.keys(aggregate).forEach(async function(rowKey) {
+      const facetData = aggregate[rowKey];
+      processedBucket
+          .file(`${rowKey}/promise_data.json`)
```

Comment:
Should this be:
```suggestion
          .file(`${rowKey}/data.json`)
```
(to match the [Python version](https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/blob/main/app/python/process.py#L132C32-L132C41)?)

Reply:
Ah yes, this was a debugging step for me (so I could compare data and promise_data to confirm they were the same)